src/Security/MemberAuthenticator/MemberLoginForm.php

```diff
@@ -131,23 +135,25 @@ protected function getFormFields()
             $backURL = $request->getSession()->get('BackURL');
         }
 
-        $label = Member::singleton()->fieldLabel(Member::config()->get('unique_identifier_field'));
+        $uniqueIdentifierFieldName = Member::config()->get('unique_identifier_field'); 
+        $label = Member::singleton()->fieldLabel($uniqueIdentifierFieldName);
+        
         $fields = FieldList::create(
             HiddenField::create("AuthenticationMethod", null, $this->getAuthenticatorClass(), $this),
             // Regardless of what the unique identifer field is (usually 'Email'), it will be held in the
             // 'Email' value, below:
             // @todo Rename the field to a more generic covering name
-            $emailField = TextField::create("Email", $label, null, null, $this),
+            $uniqueIdentifierField = ($uniqueIdentifierFieldName == 'Email' && !empty(DefaultAdminService::getDefaultAdminEmail())) ? EmailField::create("Email", $label, null, null, $this) : TextField::create($uniqueIdentifierFieldName, $label, null, null, $this), 
             PasswordField::create("Password", _t('SilverStripe\\Security\\Member.PASSWORD', 'Password'))
         );
-        $emailField->setAttribute('autofocus', 'true');
+        $uniqueIdentifierField->setAttribute('autofocus', 'true'); 
 
-        if (Security::config()->get('remember_username')) {
-            $emailField->setValue($this->getSession()->get('SessionForms.MemberLoginForm.Email'));
+        if (Security::config()->get('remember_' . strtolower($uniqueIdentifierFieldName))) { 
```

Comment:
dynamic configs are a bad idea - this will also break for all existing sites that use emails as usernames